tests fail on python 3.13

Trying to run the tests in a python 3.13 environment:

```
pytest flags: -m pytest
============================= test session starts ==============================
platform linux -- Python 3.13.2, pytest-8.3.5, pluggy-1.5.0
rootdir: /build/source
configfile: pyproject.toml
testpaths: tests
collected 35 items                                                             

tests/test_cli.py .                                                      [  2%]
tests/test_engine.py ....................F..                             [ 68%]
tests/test_session.py ...E.......                                        [100%]

==================================== ERRORS ====================================
____________________ ERROR at setup of test_default_session ____________________

cls = <class '_pytest.runner.CallInfo'>
func = <function call_and_report.<locals>.<lambda> at 0xfffff35296c0>
when = 'setup'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: Callable[[], TResult],
        when: Literal["collect", "setup", "call", "teardown"],
        reraise: type[BaseException] | tuple[type[BaseException], ...] | None = None,
    ) -> CallInfo[TResult]:
        """Call func, wrapping the result in a CallInfo.
    
        :param func:
            The function to call. Called without arguments.
        :type func: Callable[[], _pytest.runner.TResult]
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: TResult | None = func()

/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/runner.py:341: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/runner.py:242: in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
/nix/store/sm8j69rcw2q7qclragw580sb0s0srp93-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_hooks.py:513: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
/nix/store/sm8j69rcw2q7qclragw580sb0s0srp93-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py:90: in pytest_runtest_setup
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def unraisable_exception_runtest_hook() -> Generator[None]:
        with catch_unraisable_exception() as cm:
            try:
                yield
            finally:
                if cm.unraisable:
                    if cm.unraisable.err_msg is not None:
                        err_msg = cm.unraisable.err_msg
                    else:
                        err_msg = "Exception ignored in"
                    msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                    msg += "".join(
                        traceback.format_exception(
                            cm.unraisable.exc_type,
                            cm.unraisable.exc_value,
                            cm.unraisable.exc_traceback,
                        )
                    )
>                   warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Connection object at 0xfffff352d6c0>
E                   
E                   Traceback (most recent call last):
E                     File "/nix/store/92piz860vw3lki8wvs2kcpg5ibkhw9xn-python3-3.13.2/lib/python3.13/ast.py", line 290, in iter_child_nodes
E                       def iter_child_nodes(node):
E                       
E                   ResourceWarning: unclosed database in <sqlite3.Connection object at 0xfffff352d6c0>

/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning
=================================== FAILURES ===================================
_________________________ test_sqlite_memory_defaults __________________________

cls = <class '_pytest.runner.CallInfo'>
func = <function call_and_report.<locals>.<lambda> at 0xfffff4059f80>
when = 'call'
reraise = (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)

    @classmethod
    def from_call(
        cls,
        func: Callable[[], TResult],
        when: Literal["collect", "setup", "call", "teardown"],
        reraise: type[BaseException] | tuple[type[BaseException], ...] | None = None,
    ) -> CallInfo[TResult]:
        """Call func, wrapping the result in a CallInfo.
    
        :param func:
            The function to call. Called without arguments.
        :type func: Callable[[], _pytest.runner.TResult]
        :param when:
            The phase in which the function is called.
        :param reraise:
            Exception or exceptions that shall propagate if raised by the
            function, instead of being wrapped in the CallInfo.
        """
        excinfo = None
        start = timing.time()
        precise_start = timing.perf_counter()
        try:
>           result: TResult | None = func()

/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/runner.py:341: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/runner.py:242: in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
/nix/store/sm8j69rcw2q7qclragw580sb0s0srp93-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_hooks.py:513: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
/nix/store/sm8j69rcw2q7qclragw580sb0s0srp93-python3.13-pluggy-1.5.0/lib/python3.13/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/threadexception.py:92: in pytest_runtest_call
    yield from thread_exception_runtest_hook()
/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/threadexception.py:68: in thread_exception_runtest_hook
    yield
/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py:95: in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def unraisable_exception_runtest_hook() -> Generator[None]:
        with catch_unraisable_exception() as cm:
            try:
                yield
            finally:
                if cm.unraisable:
                    if cm.unraisable.err_msg is not None:
                        err_msg = cm.unraisable.err_msg
                    else:
                        err_msg = "Exception ignored in"
                    msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
                    msg += "".join(
                        traceback.format_exception(
                            cm.unraisable.exc_type,
                            cm.unraisable.exc_value,
                            cm.unraisable.exc_traceback,
                        )
                    )
>                   warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Connection object at 0xfffff408ac50>
E                   
E                   Traceback (most recent call last):
E                     File "/nix/store/d0k45jxzf24rbarnh0qn94id75jrhk54-python3.13-sqlalchemy-2.0.40/lib/python3.13/site-packages/sqlalchemy/event/base.py", line 177, in _listen
E                       def _listen(self, event_key: _EventKey[_ET], **kw: Any) -> None:
E                       
E                   ResourceWarning: unclosed database in <sqlite3.Connection object at 0xfffff408ac50>

/nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning
=========================== short test summary info ============================
FAILED tests/test_engine.py::test_sqlite_memory_defaults - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Con...
ERROR tests/test_session.py::test_default_session - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Con...
==================== 1 failed, 33 passed, 1 error in 0.23s =====================
error: builder for '/nix/store/j3dwga8nwy9cikbmzm2vsvcjr7csr01z-python3.13-flask-sqlalchemy-lite-0.1.0.drv' failed with exit code 1;
       last 25 log lines:
       >                         err_msg = cm.unraisable.err_msg
       >                     else:
       >                         err_msg = "Exception ignored in"
       >                     msg = f"{err_msg}: {cm.unraisable.object!r}\n\n"
       >                     msg += "".join(
       >                         traceback.format_exception(
       >                             cm.unraisable.exc_type,
       >                             cm.unraisable.exc_value,
       >                             cm.unraisable.exc_traceback,
       >                         )
       >                     )
       > >                   warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
       > E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Connection object at 0xfffff408ac50>
       > E                   
       > E                   Traceback (most recent call last):
       > E                     File "/nix/store/d0k45jxzf24rbarnh0qn94id75jrhk54-python3.13-sqlalchemy-2.0.40/lib/python3.13/site-packages/sqlalchemy/event/base.py", line 177, in _listen
       > E                       def _listen(self, event_key: _EventKey[_ET], **kw: Any) -> None:
       > E                       
       > E                   ResourceWarning: unclosed database in <sqlite3.Connection object at 0xfffff408ac50>
       >
       > /nix/store/5i59k97qxzm4rcg87k0qgsrzk434mwzg-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning
       > =========================== short test summary info ============================
       > FAILED tests/test_engine.py::test_sqlite_memory_defaults - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Con...
       > ERROR tests/test_session.py::test_default_session - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <sqlite3.Con...
       > ==================== 1 failed, 33 passed, 1 error in 0.23s =====================
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import collections.abc as c
+import gc
 import os
 from pathlib import Path
 
@@ -13,13 +14,14 @@ from flask_alembic import Alembic
 
 
 @pytest.fixture
-def app(request: pytest.FixtureRequest, tmp_path: Path) -> Flask:
+def app(request: pytest.FixtureRequest, tmp_path: Path) -> c.Iterator[Flask]:
     app = Flask(request.module.__name__, root_path=os.fspath(tmp_path))
     app.config["SQLALCHEMY_ENGINES"] = {
         "default": "sqlite://",
         "other": "sqlite://",
     }
-    return app
+    yield app
+    gc.collect()
 
 
 @pytest.fixture
@@ -29,8 +31,13 @@ def app_ctx(app: Flask) -> c.Iterator[AppContext]:
 
 
 @pytest.fixture
-def db(app: Flask) -> SQLAlchemy:
-    return SQLAlchemy(app)
+def db(app: Flask) -> c.Iterator[SQLAlchemy]:
+    db = SQLAlchemy(app)
+    yield db
+
+    with app.app_context():
+        for engine in db.engines.values():
+            engine.dispose()
 
 
 @pytest.fixture

--- a/tests/test_fsa.py
+++ b/tests/test_fsa.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import collections.abc as c
 import os
 import typing as t
 from pathlib import Path
@@ -13,12 +14,17 @@ from flask_alembic import Alembic
 
 
 @pytest.fixture
-def db(app: Flask) -> SQLAlchemy:
+def db(app: Flask) -> c.Iterator[SQLAlchemy]:
     app.config["SQLALCHEMY_BINDS"] = {
         None: "sqlite://",
         "other": "sqlite://",
     }
-    return SQLAlchemy(app)
+    db = SQLAlchemy(app)
+    yield db
+
+    with app.app_context():
+        for engine in db.engines.values():
+            engine.dispose()
 
 
 @pytest.fixture
@@ -64,3 +70,5 @@ def test_override_engines(tmp_path: Path, app: Flask, db: SQLAlchemy) -> None:
     alembic = Alembic(app, engines=engine)
     assert alembic.migration_context.connection is not None
     assert alembic.migration_context.connection.engine.url.database == db_path
+    alembic.migration_context.connection.close()
+    engine.dispose()

--- a/tests/test_fsa_lite.py
+++ b/tests/test_fsa_lite.py
@@ -12,20 +12,6 @@ from sqlalchemy.orm import DeclarativeBase
 from flask_alembic import Alembic
 
 
-@pytest.fixture
-def app(app: Flask) -> Flask:
-    app.config["SQLALCHEMY_ENGINES"] = {
-        "default": "sqlite://",
-        "other": "sqlite://",
-    }
-    return app
-
-
-@pytest.fixture
-def db(app: Flask) -> SQLAlchemy:
-    return SQLAlchemy(app)
-
-
 class Model(DeclarativeBase):
     pass
 
@@ -54,11 +40,14 @@ def test_uses_engines(app: Flask) -> None:
 def test_engine_required(app: Flask) -> None:
     """Flask-SQLAlchemy-Lite must configure an engine."""
     del app.config["SQLALCHEMY_ENGINES"]
-    SQLAlchemy(app, require_default_engine=False)
+    db = SQLAlchemy(app, require_default_engine=False)
     alembic = Alembic(app, metadatas=Model.metadata)
 
     with pytest.raises(RuntimeError, match="engines configured"):
         assert alembic.migration_context
+
+    for engine in db.engines.values():
+        engine.dispose()
 
 
 @pytest.mark.usefixtures("app_ctx", "db")
@@ -80,3 +69,5 @@ def test_override_engines(tmp_path: Path, app: Flask) -> None:
     alembic = Alembic(app, metadatas=Model.metadata, engines=engine)
     assert alembic.migration_context.connection is not None
     assert alembic.migration_context.connection.engine.url.database == db_path
+    alembic.migration_context.connection.close()
+    engine.dispose()

--- a/tests/test_plain_sa.py
+++ b/tests/test_plain_sa.py
@@ -45,6 +45,8 @@ def test_valid_config(app: Flask) -> None:
         engines={"default": engine, "other": other_engine},
     )
     assert len(alembic.migration_contexts) == 2
+    engine.dispose()
+    other_engine.dispose()
 
 
 @pytest.mark.usefixtures("app_ctx")
@@ -59,3 +61,5 @@ def test_missing_engine(app: Flask) -> None:
 
     with pytest.raises(RuntimeError, match="Missing engine config"):
         assert alembic.migration_context
+
+    engine.dispose()

